hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -386,6 +398,25 @@ public DatanodeDeletedBlockTransactions getTransactions(
             LOG.warn("Container: {} was not found for the transaction: {}.", id, txn);
             txIDs.add(txn.getTxID());
           }
+          if (!iter.hasNext()) {
+            try (TableIterator<Long,
+                ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> tmpIter =
+                deletedBlockLogStateManager.getReadOnlyIterator()) {
+              if (tmpIter.hasNext()) {
+                keyValue = tmpIter.next();
+                if (keyValue.getKey() != firstProcessedTransactionKey) {
+                  iter.seek(keyValue.getKey());
+                } else {
+                  break;
+                }
+              }
+            }
```

Comment:
```suggestion
            iter.seekToFirst();
```

Reply:
@nandakumar131 Thanks for the suggestion, handled it.